components/Modal/Modal.js

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import { useEffect } from "react";
+import clsx from "clsx";
+import { createPortal } from "react-dom";
+
+import styles from "./Modal.module.css";
+
+export default function Modal({ open, close, closeOnOutsideClick = true, children, className }) {
```

Comment:
maybe rename close to `onClose`/`onDismiss` to make it clear it's a function to call when the modal's closed (if i'm right about that). and maybe `closeOnOutsideClick` => `dismissable` or `persistent`?